runtime-config.md

```diff
@@ -22,6 +23,13 @@ In Linux, for e.g., the network namespace could be configured in this hook.
 
 If a hook returns a non-zero exit code, then an error including the exit code and the stderr is returned to the caller and the container is torn down.
 
+### Post-start
+
+The post-start hooks are called after the user process is started.
+It can be used for notifying user that real process is spawned.
```

Comment:
On Tue, Sep 29, 2015 at 02:20:06PM -0700, Alexander Morozov wrote:
> > +The post-start hooks are called after the user process is started.
> > +It can be used for notifying user that real process is spawned.
> 
> There is nothing about "Sidecar application" as well as "sidecar
> process" in google, so seems like not so familiar language…

Ah, it looks like the language I was reaching for is “sidekick” [1],
but anyway fair enough on it not being well established.

> … cloudfoundry use-case isn't sidecar, but notfiy use-case indeed.

Ah, good catch.  But it's still not the notificiation that's special
about the post-start timing (it's the fact that the container is ready
for additional execs).  So I'd rather have something like:

  For example, this hook could launch additional applications joining
  this container.

or if you really want to echo the cloudfoundary use case:

  For example, this hook could tell an external manager that it was
  safe to launch additional applications joining this container.

I intend to do the former and launch a dummy process with start, then
post-start (or whatever "this container is ready to exec into" tooling
we get) to launch Nginx inside that container.  Decoupling the Nginx
application from the main container application lets me upgrade my
binary [2] without killing the container.

[1]: https://www.digitalocean.com/community/tutorials/how-to-use-confd-and-etcd-to-dynamically-reconfigure-services-in-coreos
[2]: https://www.nginx.com/resources/wiki/start/topics/tutorials/commandline/#upgrading-to-a-new-binary-on-the-fly